test/GroupByTest.cpp

```diff
@@ -1912,3 +1911,282 @@ TEST(GroupBy, Descriptor) {
+  GroupBy groupBy{qec_, {V{"?x"}}, {}, std::move(subtree)};
+
+  expectReturningIdTables<0>(groupBy, {});
+}
+
+// _____________________________________________________________________________
+TEST_P(GroupByLazyFixture, verifyEmptyIdTableIsSkipped) {
+  // SELECT (SUM(?x) as ?sum) WHERE { ... } GROUP BY ?y
+  std::vector<IdTable> idTables;
+  idTables.emplace_back(2, qec_->getAllocator());
+  idTables.push_back(makeIdTableFromVector({{1, 0}}, IntId));
+  idTables.emplace_back(2, qec_->getAllocator());
+  idTables.emplace_back(2, qec_->getAllocator());
+  idTables.push_back(makeIdTableFromVector({{2, 0}}, IntId));
+  idTables.emplace_back(2, qec_->getAllocator());
+  idTables.push_back(makeIdTableFromVector({{4, 1}}, IntId));
+  idTables.emplace_back(2, qec_->getAllocator());
+
+  auto subtree = makeExecutionTree<ValuesForTesting>(
+      qec_, std::move(idTables), vars("?x", "?y"), true,
+      std::vector<ColumnIndex>{1});
+  Alias alias{SparqlExpressionPimpl{makeSum("?x"), "SUM(?x)"}, V{"?sum"}};
+  GroupBy groupBy{qec_, {V{"?y"}}, {std::move(alias)}, std::move(subtree)};
+
+  expectReturningIdTables<2>(groupBy, {makeIdTableFromVector({{0, 3}}, IntId),
+                                       makeIdTableFromVector({{1, 4}}, IntId)});
+}
+
+// _____________________________________________________________________________
+TEST_P(GroupByLazyFixture, verifyImplicitGroupByWorksWithNonEmptyInput) {
+  std::vector<IdTable> idTables;
+  idTables.push_back(makeIdTableFromVector({{1}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{2}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{4}}, IntId));
+
+  auto subtree = makeExecutionTree<ValuesForTesting>(
+      qec_, std::move(idTables), vars("?x"), true, std::vector<ColumnIndex>{0});
+  Alias alias{SparqlExpressionPimpl{makeSum("?x"), "SUM(?x)"}, V{"?sum"}};
+  GroupBy groupBy{qec_, {}, {std::move(alias)}, std::move(subtree)};
+
+  expectReturningIdTables<1>(groupBy, {makeIdTableFromVector({{7}}, IntId)});
+}
+
+// _____________________________________________________________________________
+TEST_P(GroupByLazyFixture,
+       verifyGroupsSpanningOverMultipleIdTablesAreAggregatedCorrectly) {
+  std::vector<IdTable> idTables;
+  idTables.push_back(makeIdTableFromVector({{1, 0}, {2, 1}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{4, 1}, {8, 1}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{16, 1}, {32, 2}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{64, 2}, {128, 2}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{256, 3}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{512, 3}}, IntId));
+
+  auto subtree = makeExecutionTree<ValuesForTesting>(
+      qec_, std::move(idTables), vars("?x", "?y"), true,
+      std::vector<ColumnIndex>{1});
+  Alias alias{SparqlExpressionPimpl{makeSum("?x"), "SUM(?x)"}, V{"?sum"}};
+  GroupBy groupBy{qec_, {V{"?y"}}, {std::move(alias)}, std::move(subtree)};
```

Comment:
As you always do the same expression (two inputs, x and y, SUM(x), Group By Y you can also abstract away some of this
redundant initialization (you can basically turn the `idTables` into a `GroupBy`
`GroupBy groupBy = makeSumXGroupByY(std::move(idTables));`.